src/compression/snappy.rs

```diff
@@ -1,18 +1,37 @@
 use std::io::{self, Read};
 
 use byteorder::{BigEndian, ByteOrder};
-use snappy;
+use snap;
 
 use error::{Result, Error};
 
 pub fn compress(src: &[u8]) -> Result<Vec<u8>> {
-    Ok(snappy::compress(src))
+    let mut buf = vec![0; snap::max_compress_len(src.len())];
+
+    snap::Encoder::new()
+        .compress(src, &mut buf)
+        .map(|len| {
+                 buf.truncate(len);
```

Comment:
yes, i think the `truncate/set_len` is necessary here. after all, the size of the initial allocation of `buf` is merely an estimate (upper bound).